proposals/lip-0061.md

```diff
@@ -564,14 +571,21 @@ def verifyAggregateCommit(blockHeader: BlockHeader)-> bool:
     # maxHeightPrecommitted before processing blockHeader and updating the BFT store.
     if aggregateCommit.height > getBFTHeights().maxHeightPrecommitted:
         return False
+    # The heights of aggregate commits must be at least MIN_CERTIFY_HEIGHT.
+    if aggregateCommit.height < MIN_CERTIFICATE_HEIGHT:
+        return False
     # If there are new BFT parameters for a height h, then the chain needs to include
     # an aggregate commit for the block at height h-1. This block and the corresponding
     # certificate authenticate the BFT parameters via the validators hash property.
     # Note that the BFT parameters at height maxHeightCertified+1 are already authenticated
     # by the previous certificate.
-    heightNextBFTParameters = getNextHeightBFTParameters(getBFTHeights().maxHeightCertified+1)
-    if previous function call returns valid height and aggregateCommit.height > heightNextBFTParameters-1:
-        return False
+    try:
+        heightNextBFTParameters = getNextHeightBFTParameters(getBFTHeights().maxHeightCertified+1)
+        if heightNextBFTParameters > MIN_CERTIFICATE_HEIGHT 
```

Comment:
I am a bit unsure about this extra condition. I would have said that we should check `aggregateCommit.height > MIN_CERTIFICATE_HEIGHT` (which you did above). Here we are saying that the aggregate commit can be valid if, for instance, `aggregateCommit.height == MIN_CERTIFICATE_HEIGHT >= heightNextBFTParameters`.

Reply:
This try- except here has to do with chain of trust verification. So, if we ignore `MIN_CERTIFY_HEIGHT` (say it has value -1), what we do is to enforce chain of trust by saying "for every change of bft parameters at height `h`, need to include a certificate for height `h-1`".

When `MIN_CERTIFY_HEIGHT` comes into the game, we want to say "for every change of bft parameters at height `h > MIN_CERTIFICATE_HEIGHT`, need to include a certificate for height `h-1`" . The reason is that for changes of bft params at height `h <= MIN_CERTIFICATE_HEIGHT`, we can not have a certificate at height `h-1`..

Does it make sense?

Reply:
In particular for the example that you give: Assume the first certificate is produced for height `h = MIN_CERTIFICATE_HEIGHT`, i.e., `aggregateCommit.height = MIN_CERTIFICATE_HEIGHT`. 

Then, I guess this certificate should be accepted and it would be the first one produced. After this one, chain of trust will be enforced.